pkg/web/router/router.go

```diff
@@ -152,6 +161,53 @@ func SetAppRoutes(r *echo.Group, appsHandler apps.HTTPHandler) {
 	//   404:
 	//     description: App not found
 	r.POST("/apps/:id/versions/disable", appsHandler.DisableAllAppVersionsByAppID)
+
+	// Create an app
+	// ---
+	// summary:
+	// - Create an new app
+	// - Update the deleted_at field of an app when the appId already exist
+	// operationId: CreateApp
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: body
+	//   in: body
+	//   description:
+	//   required: true
+	//   schema:
+	//     $ref: '#/definitions/App'
+	// responses:
+	//   201:
+	//     description: successful operation
+	//   400:
+	//     description: Invalid data supplied
+	r.POST("/apps", appsHandler.CreateApp)
+
+	// Update an app
+	// ---
+	// summary: Update the app name of an app
+	// operationId: UpdateAppNameByID
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: id
+	//   in: path
+	//   description: The id for the app that will have all its versions updated
+	//   required: true
+	//   type: string
+	// - name: body
+	//   in: body
+	//   description:
+	//   required: true
+	//   schema:
+	//     $ref: '#/definitions/App'
+	// responses:
+	//   201:
+	//     description: successful operation
+	//   400:
+	//     description: Invalid data supplied
+	r.PATCH("/apps/:id", appsHandler.UpdateAppNameByID)
```

Comment:
A PATCH on a resource should technically allow the user to update any field for an app resource, not just the single app name field. For now this will be ok. We could make it more generic in the future.

Reply:
But, if it should allow updating any field then shows that it would should be a PUT and not a PATCH.

Reply:
A PUT would be updating the entire resource. PATCH allow you to update 1-many fields within a resource.

This implementation of PATCH only allows the user to update the app name. It is not generic so that if we added new fields to the resource that it would work automatically.

Reply:
We don't have other fields by the business roles that we can update anyway. Just the name and deleted_at. However, I understand that the deleted_at should not be updated here too.